### ISSUE TITLE
Incompatible with Rubinius

### ISSUE THREAD

**Issue by paddor** [guard/listen#391]
When Bundler tries to upgrade listen from 3.0.5 to 3.1.3, i get the following error:

```
Gem::InstallError: listen requires Ruby version >= 2.2.3, ~> 2.2.
Using rspec 3.4.0
An error occurred while installing listen (3.1.3), and Bundler cannot continue.
Make sure that `gem install listen -v '3.1.3'` succeeds before bundling.
```

I'm using Rubinius 3.30.

What exactly do you mean by "Works Everywhere!" in your gem's description?

**Comment by e2** [comment]
Hi @paddor , thanks for reporting this.

Rubinius isn't guaranteed to work. I never actually even got the Travis tests working properly, and since it's never tested, it's a bit of a stretch to tell people it's supported.

You can lock to Listen `~> 3.0.x` to get patches for any major bug fixes.

If someone can submit a PR with a working RBX build on Travis, that would help. I can help get it working, but right now even JRuby tests don't pass, so I believe it needs some patches there too.

I'd like to get Rubinius working, but I simply don't have time to check out why the build used to Fail (you can check Travis build logs for earlier versions if you're curious).

By "Works Everywhere!" it was supposed to mean OSs (OSX, Linux, Windows, BSD).

I'd be happy to accept patches - for now I want to "guarantee" support for some scenarios, and that means marking "unknown" scenarios (e.g. Rubinius) as "not supported (yet)".

It just reflects reality. It doesn't install because it's not expected to work (although it could, but that's unknown because tests have always failed).

I admit I "cheated" with JRuby - I added it to make it installable just on 9.0.5.0 since that version is the last released that isn't affected by a known security vulnerability.

Listen makes heavy use of Threads, descriptors and filesystem API during tests - and all three combined can be a time sink to properly handle on both JRuby and Rubinius.

If there's a released version of RBX supporting Ruby 2.3, I'd be happy to include it as "supported and thus installable" as long as the tests "mostly" work.

I'm very happy to help get a PR merged with any fixes for this and release an updated version of Listen with Rubinius support! (I've simply been drained of time by open source work recently).

**Comment by paddor** [comment]
Thanks, I locked it to `~> 3.0.x`.
